app/models/mixins/retirement_mixin.rb

```diff
@@ -253,15 +253,18 @@ def valid_zone?
   end
 
   def system_context_requester
-    if try(:evm_owner_id).present?
-      if User.find_by(:id => evm_owner_id).present?
-        return evm_owner
-      else
-        $log.info("#{name} has evm_owner_id present but no user with that id found so defaulting to admin.")
-        return User.super_admin
+    user = User.find_by(:id => evm_owner_id) if try(:evm_owner_id)
+    return evm_owner if user.present? && user.current_group.present?
+
+    if user.present?
+      msg = "#{name} has evm_owner present"
+      if user.current_group.blank?
+        msg += " but user has no group so defaulting to admin."
       end
+    else
+      msg = "System context defaulting to admin user because owner of #{name} not set."
     end
```

Comment:
Last time I refactored this according to existing logic. 
After second thought this can be simplified to:
```
if user.present?
  msg = "#{name} has evm_owner present but user has no group so defaulting to admin."
else
  msg = "System context defaulting to admin user because owner of #{name} not set."
end
```

Reply:
Fine, except it needs the group check too.

Reply:
Group check has been done when the code reaches this point.